Dockerfile

```diff
@@ -1,19 +1,54 @@
-FROM golang:1.22
+# The base image for the supersim image can be modified
+# by specifying BASE_IMAGE build argument
+ARG BASE_IMAGE=golang:1.22
 
-RUN apt-get update && apt-get install -y curl git
+# 
+# This stage builds the foundry binaries
+# 
```

Comment:
for the comments on the different stages it might be better if we add comments like this https://github.com/ethereum-optimism/ecosystem/blob/ef2845eaecf99581be17c2020fb53e05096e94da/Dockerfile#L10-L12 so it's easier to see the different stages at a glance

Reply:
@nitaliano In my dockerfiles I usually go even crazier, just wasn't sure whether you'd like it. [Here's an example](https://github.com/LayerZero-Labs/devtools/blob/main/Dockerfile#L35), what do you think?

```yaml
#   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-
#  / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \
# `-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'
#
#          This stage builds the foundry binaries
#
#   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-
#  / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \
# `-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'
```

Reply:
haha nice! yea totally that works as well just something that helps the developer understand how many stages exist at a glance